src/comparator/file_set_comparator.py

```diff
@@ -218,6 +218,12 @@ def _compare_resources(self, fs_original, fs_update):
                 message=f"A new resource definition `{resource_type}` has been added.",
                 change_type=ChangeType.MINOR,
             )
-        # 3. File-level resource definitions removal may not be breaking change since
-        # the resource could be moved to message-level. This will be checked in the message
-        # and field level (where the resource is referenced).
+        # 3. File-level resource definitions removal.
+        for resource_type in resources_types_original - resources_types_update:
```

Comment:
All resources are registered in this database, so any resource that is removed should be considered breaking.